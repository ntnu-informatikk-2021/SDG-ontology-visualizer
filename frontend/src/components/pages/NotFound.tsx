import { ArrowLeftIcon } from '@chakra-ui/icons';
import { Button, Heading, Stack, Text } from '@chakra-ui/react';
import React from 'react';
import { useHistory } from 'react-router-dom';

const NotFound: React.FC = () => {
  const history = useHistory();

  return (
    <Stack spacing="10" justify="center" align="center" color="white" h="20em" bg="cyan.800">
      <Heading as="h2" size="3xl">
        404
      </Heading>
      <Text fontSize="xl">Denne siden eksisterer ikke</Text>
      <Button
        colorScheme="white-alpha"
        bg="white"
        color="cyan.800"
        leftIcon={<ArrowLeftIcon />}
        onClick={() => {
          history.push('/');
        }}
      >
        Tilbake til forsiden
      </Button>
    </Stack>
  );
};

export default NotFound;
