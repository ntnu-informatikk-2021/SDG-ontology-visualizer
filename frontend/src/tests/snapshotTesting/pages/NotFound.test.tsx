import React from 'react';
import renderer from 'react-test-renderer';

import NotFound from '../../../components/pages/NotFound';

it('renders when there are no items', () => {
  const tree = renderer.create(<NotFound />).toJSON();
  expect(tree).toMatchSnapshot();
});
