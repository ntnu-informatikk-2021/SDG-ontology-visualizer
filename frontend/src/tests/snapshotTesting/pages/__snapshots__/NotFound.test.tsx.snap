// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no items 1`] = `
<div
  className="chakra-stack css-19g0xa"
>
  <h2
    className="chakra-heading css-0"
  >
    404
  </h2>
  <p
    className="chakra-text css-6dvxm2"
  >
    Denne siden eksisterer ikke
  </p>
  <button
    className="chakra-button css-1c5reui"
    onClick={[Function]}
    type="button"
  >
    <span
      className="chakra-button__icon css-ktvjz1"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-onkibi"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <g
          fill="currentColor"
        >
          <path
            d="M10.416,12a2.643,2.643,0,0,1,.775-1.875L20.732.584a1.768,1.768,0,0,1,2.5,2.5l-8.739,8.739a.25.25,0,0,0,0,.354l8.739,8.739a1.768,1.768,0,0,1-2.5,2.5l-9.541-9.541A2.643,2.643,0,0,1,10.416,12Z"
          />
          <path
            d="M.25,12a2.643,2.643,0,0,1,.775-1.875L10.566.584a1.768,1.768,0,0,1,2.5,2.5L4.327,11.823a.25.25,0,0,0,0,.354l8.739,8.739a1.768,1.768,0,0,1-2.5,2.5L1.025,13.875A2.643,2.643,0,0,1,.25,12Z"
          />
        </g>
      </svg>
    </span>
    Tilbake til forsiden
  </button>
</div>
`;
